src/routes/post-article.ts

```diff
@@ -0,0 +1,22 @@
+import { constants } from 'http2';
+import { Context, Middleware, Next } from 'koa';
+import { JsonLdObj } from 'jsonld/jsonld-spec';
+import uniqueString from 'unique-string';
+import Articles from '../articles';
+import ArticleIdAlreadySet from '../errors/article-id-already-set';
+
+export default (articles: Articles): Middleware => (
+  async ({ request, response }: Context, next: Next): Promise<void> => {
+    const article = request.body as JsonLdObj;
+    if ('@id' in article) {
+      throw new ArticleIdAlreadySet(article['@id']);
```

Comment:
Been thinking about this. `409 Conflict` makes sense if an article with that ID is already in the collection, but really we're saying you're not allowed to set an ID, so `403 Forbidden` covers it. Suggest we also throw an HTTP error directly here for simplicity.

Reply:
I've made this refactor.